### PR TITLE
Fixed some issues with using grommet in a shadow root

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Addresses some issues with using grommet inside a shadow root context. Specifically, too much reliance on `document` being the root of things.

Also, added a "document" vs. "shadow" toolbar control in storybook to allow testing any story from a shadow root context.

#### Where should the reviewer start?

DropContainer.js

#### What testing has been done on this PR?

Various stories with the new shadow root context, certainly not all though.

#### How should this be manually tested?

stories with shadow root context

#### Do Jest tests follow these best practices?

no test changes

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

**Comment by taysea** [comment]
I like how in the current storybook, the theme selector says "Theme" -- I was expecting something similar in the deploy that said "Root". I noticed it's changed to say "grommet" and "document" which I didn't immediately recognize as a selector. soft opinion.

Screen Shot 2023-03-21 at 10 05 50 AM

NOW
Screen Shot 2023-03-21 at 10 09 29 AM

File: src/js/components/Layer/Layer.js [hunk 1]
```diff
@@ -54,10 +54,9 @@ const Layer = forwardRef((props, ref) => {
           }
           setTimeout(() => {
             // we add the id and query here so the unit tests work
-            const clone =
-              containerTarget === document.body
-                ? document.getElementById('layerClone')
-                : containerTarget.getElementById('layerClone');
+            const clone = containerTarget
+              .getRootNode()
```
Comment (taysea) [inline src/js/components/Layer/Layer.js:58]: Will this ever be undefined?
```suggestion
              .getRootNode()?
```

Comment (ericsoderberghp) [reply src/js/components/Layer/Layer.js:58]: I don't think so: https://developer.mozilla.org/en-US/docs/Web/API/Node/getRootNode#return_value

**Comment by ericsoderberghp** [comment]
> I like how in the current storybook, the theme selector says "Theme" -- I was expecting something similar in the deploy that said "Root". I noticed it's changed to say "grommet" and "document" which I didn't immediately recognize as a selector. soft opinion.
> 
> Screen Shot 2023-03-21 at 10 05 50 AM
> 
> NOW Screen Shot 2023-03-21 at 10 09 29 AM

changed as suggested, I was soft too :)

**Review by taysea — APPROVED** [review]
LGTM.

**Review by MikeKingdom — APPROVED** [review]
Sorry for the delay. Looks great.

**Pull request merged**